elide-core/src/main/java/com/yahoo/elide/core/security/executors/AggregationStorePermissionExecutor.java

```diff
@@ -0,0 +1,165 @@
+     * Aggregation Datastore model can only have user checks at field level permission expression.
+     * @param resource resource
+     * @param changeSpec changepsec
+     * @param annotationClass annotation class
+     * @param field field to check
+     * @param <A>
+     * @return
+     */
+    @Override
+    public <A extends Annotation> ExpressionResult checkSpecificFieldPermissions(PersistentResource<?> resource,
+                                                                                 ChangeSpec changeSpec,
+                                                                                 Class<A> annotationClass,
+                                                                                 String field) {
+        if (!annotationClass.equals(ReadPermission.class)) {
+            return ExpressionResult.FAIL;
+        }
+        return checkUserPermissions(resource.getResourceType(), annotationClass, field);
+    }
+
+    /**
+     * Not supported in aggregation datastore.
+     * @param resource resource
+     * @param changeSpec changepsec
+     * @param annotationClass annotation class
+     * @param field field to check
+     * @param <A>
+     * @return
+     */
+    @Override
+    public <A extends Annotation> ExpressionResult checkSpecificFieldPermissionsDeferred(PersistentResource<?> resource,
+                                                                                         ChangeSpec changeSpec,
+                                                                                         Class<A> annotationClass,
+                                                                                         String field) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Check strictly user permissions on an entity and any of the requested field. Evaluates
+     * expression = (entityRule AND (field1Rule OR field2Rule ... OR fieldNRule))
+     *
+     * @param <A> type parameter
+     * @param resourceClass Resource class
+     * @param annotationClass Annotation class
+     */
+    @Override
+    public <A extends Annotation> ExpressionResult checkUserPermissions(Type<?> resourceClass,
+                                                                        Class<A> annotationClass,
+                                                                        Set<String> requestedFields) {
+        if (!annotationClass.equals(ReadPermission.class)) {
+            return ExpressionResult.FAIL;
+        }
+
+        Supplier<Expression> expressionSupplier = () ->
+                expressionBuilder.buildUserCheckEntityAndAnyFieldExpression(
+                        resourceClass,
+                        annotationClass,
+                        requestedFields,
+                        requestScope);
+
+        return checkOnlyUserPermissions(
+                resourceClass,
+                annotationClass,
+                Optional.empty(),
```

Comment:
I found a bug in the user check cache in Elide 5.  The user check cache will look at either one field or no fields (this case).  However, this check could be called multiple times with different fields.  We really ought to make the cache always map to a set of requested fields.  Optional.empty() isn't right here, but we can't do the right thing until this bug is fixed.